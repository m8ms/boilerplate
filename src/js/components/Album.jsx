import React from 'react';
import Slider from  'react-slick';

import ImageSection from  './ImageSection';
import Comments from  './Comments';

class Album extends React.Component {

    static defaultProps = {
        album: {}
    };


    printAlbumTitle(){

        if (this.props.album.title){
            return <h3 className="album__title">{this.props.album.title}</h3>;
        }
        return '';
    }

    printAlbumAuthor(){

        if (this.props.album.account_url){
            return <div className="album__author"> by {this.props.album.account_url}</div>;
        }
        return '';
    }

    printImages(){

        if (this.props.album.images){
            return this.props.album.images.map(img => <ImageSection key={img.id} item={img}/>);
        }
    }

    getImagesCount() {
        return this.props.album.images && this.props.album.images.length || ' ';
    }

    render() {

        const sliderSettings = {
            dots: true,
            speed: 500,
            adaptiveHeight: true,
            slidesToShow: 1,
            slidesToScroll: 1
        };

        return (
            <div className="album">
                <header className="album__header">
                    {this.printAlbumTitle()}
                    {this.printAlbumAuthor()}
                    <div className="album__image-count">Images in album: {this.getImagesCount()}</div>
                </header>
                <Slider {...sliderSettings}>
                    {this.printImages()}
                </Slider>
                <Comments comments={this.props.comments}/>
            </div>

        )
    }
}

export default Album;
