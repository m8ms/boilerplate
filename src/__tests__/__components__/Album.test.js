import React from 'react';
import Album from '../../js/components/Album';
import renderer from 'react-test-renderer';
import { commentWithChildren, comment } from '../__fixtures__/comments.fixtures';
import { album } from '../__fixtures__/album.fixtures';

window.matchMedia = window.matchMedia || function() {
        return {
            matches : false,
            addListener : function() {},
            removeListener: function() {}
        };
    };

it('renders correctly with no data', () => {
    const tree = renderer
        .create(<Album/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with album', () => {
    const tree = renderer
        .create(<Album album={album}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with album and comments', () => {
    const tree = renderer
        .create(<Album album={album} comments={[commentWithChildren, comment]}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
