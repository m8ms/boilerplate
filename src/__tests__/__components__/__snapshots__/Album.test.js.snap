// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with album 1`] = `
<div
  className="album"
>
  <header
    className="album__header"
  >
    <h3
      className="album__title"
    >
      Rovings the smerr of fresh air - A Polandball Comic
    </h3>
    <div
      className="album__author"
    >
       by 
      ThatOneGeographyNerd
    </div>
    <div
      className="album__image-count"
    >
      Images in album: 
      1
    </div>
  </header>
  <div
    className="slick-slider slick-initialized"
  >
    
    <div
      className="slick-list"
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="image-section"
            >
              <div
                className="image-section__image-wrap"
              >
                <img
                  className="image-section__image"
                  src="https://i.imgur.com/GRDXXdF.png"
                />
              </div>
              <div
                className="image-section__image-meta"
              >
                <div
                  className="image-section__title"
                />
                <div
                  className="image-section__description"
                >
                  Sorry I haven't been uploading stuff like this lately. Hopefully this'll make up for it.
                </div>
                <div
                  className="image-section__views-count"
                >
                  3965
                   views
                </div>
                <div
                  className="image-section__date"
                >
                  April 8, 2018 11:49 PM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    
    
  </div>
</div>
`;

exports[`renders correctly with album and comments 1`] = `
<div
  className="album"
>
  <header
    className="album__header"
  >
    <h3
      className="album__title"
    >
      Rovings the smerr of fresh air - A Polandball Comic
    </h3>
    <div
      className="album__author"
    >
       by 
      ThatOneGeographyNerd
    </div>
    <div
      className="album__image-count"
    >
      Images in album: 
      1
    </div>
  </header>
  <div
    className="slick-slider slick-initialized"
  >
    
    <div
      className="slick-list"
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      >
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <div
              className="image-section"
            >
              <div
                className="image-section__image-wrap"
              >
                <img
                  className="image-section__image"
                  src="https://i.imgur.com/GRDXXdF.png"
                />
              </div>
              <div
                className="image-section__image-meta"
              >
                <div
                  className="image-section__title"
                />
                <div
                  className="image-section__description"
                >
                  Sorry I haven't been uploading stuff like this lately. Hopefully this'll make up for it.
                </div>
                <div
                  className="image-section__views-count"
                >
                  3965
                   views
                </div>
                <div
                  className="image-section__date"
                >
                  April 8, 2018 11:49 PM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    
    
  </div>
  <div
    className="comments"
  >
    <h3
      className="comments__header"
    >
      Comments
    </h3>
    <div
      className="comments__list"
    >
      <div
        className="comment"
      >
        <div
          className="comment__author"
        >
          <div
            className="comment__username"
          >
            whalesurfertoo
          </div>
          <div
            className="comment__points"
          >
            5pts
          </div>
          <div
            className="comment__date"
          >
            14 days ago
          </div>
        </div>
        <div
          className="comment__content"
        >
          Always upvote Poland ball
        </div>
        <div
          className="comment__sub-comments"
        >
          <div
            className="comment comment--sub-comment"
          >
            <div
              className="comment__author"
            >
              <div
                className="comment__username"
              >
                ThatOneGeographyNerd
              </div>
              <div
                className="comment__points"
              >
                5pts
              </div>
              <div
                className="comment__date"
              >
                14 days ago
              </div>
            </div>
            <div
              className="comment__content"
            >
              Unless it has advanced tools. Then no. xd
            </div>
            <div
              className="comment__sub-comments"
            >
              <div
                className="comment comment--sub-comment"
              >
                <div
                  className="comment__author"
                >
                  <div
                    className="comment__username"
                  >
                    whalesurfertoo
                  </div>
                  <div
                    className="comment__points"
                  >
                    1pt
                  </div>
                  <div
                    className="comment__date"
                  >
                    14 days ago
                  </div>
                </div>
                <div
                  className="comment__content"
                >
                  True
                </div>
                <div
                  className="comment__sub-comments comment__sub-comments--empty"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="comment"
      >
        <div
          className="comment__author"
        >
          <div
            className="comment__username"
          >
            SeaPrince
          </div>
          <div
            className="comment__points"
          >
            1pt
          </div>
          <div
            className="comment__date"
          >
            21 days ago
          </div>
        </div>
        <div
          className="comment__content"
        >
          I think Mexico likes Canada in this situation and called the US bumfuckers.
        </div>
        <div
          className="comment__sub-comments comment__sub-comments--empty"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly with no data 1`] = `
<div
  className="album"
>
  <header
    className="album__header"
  >
    
    
    <div
      className="album__image-count"
    >
      Images in album: 
       
    </div>
  </header>
  <div
    className="slick-slider slick-initialized"
  >
    
    <div
      className="slick-list"
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      />
    </div>
    
    
  </div>
</div>
`;
